boards: stm32: Fix boards names in yaml and/or doc files

erwango opened the pull request:
For atuo doc generation purpose, get name value of boards' yaml files
in sync with name provided as board name in .rst file

Fixes #24927

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32: Fix boards names in yaml and/or doc files

For auto doc generation purpose, get name value of boards' yaml files
in sync with name provided as board name in .rst file


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented on boards/arm/stm32g0316_disco/doc/index.rst:
```diff
@@ -1,6 +1,6 @@
 
-STM32G0316-DISCO
+ST STM32G0316 Discovery
 ################
```
Need to fix the hash's here

erwango replied:
The change was in the editor buffer. Fixed, txs!

erwango commented:
@nashif, fixed and updated board names.

A question on <board>/ folders with 2 targets (for instance dual core stm32h747i_disco):
There is one yaml file per target, but only one overall doc.
Is that ok to have same name yaml property for both targets ?

nashif commented:
> Is that ok to have same name yaml property for both targets ?

I guess yes, need to look how we can improve this in the future, for now however it should work.

erwango commented:
> I guess yes, need to look how we can improve this in the future, for now however it should work.

Ok, I did the opposite. Let me update.

galak commented:
Since this touches some many ST boards CI will timeout, looking at looks everything is building so going to merge by hand.
